src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/takemanagement/view/RecordableFragment.kt

```diff
@@ -0,0 +1,202 @@
+            }
+        })
+        add(dragContainer)
+    }
+
+    private fun Parent.addDragTakeEventHandlers() {
+        addEventHandler(StartDragEvent.START_DRAG, ::startDrag)
+        addEventHandler(AnimateDragEvent.ANIMATE_DRAG, ::animateDrag)
+        addEventHandler(CompleteDragEvent.COMPLETE_DRAG, ::completeDrag)
+    }
+
+    private fun Parent.addButtonEventHandlers() {
+        addEventHandler(PlayOrPauseEvent.PLAY) {
+            lastPlayOrPauseEvent.set(it)
+        }
+        addEventHandler(DeleteTakeEvent.DELETE_TAKE) {
+            recordableViewModel.deleteTake(it.take)
+        }
+        addEventHandler(EditTakeEvent.EDIT_TAKE) {
+            recordableViewModel.editTake(it)
+        }
+    }
+
+    private fun createSnackBar(pane: Pane) {
+        // TODO: This doesn't actually handle anything correctly. Need to know whether the user
+        // TODO... hasn't selected an editor or recorder and respond appropriately.
+        val snackBar = JFXSnackbar(pane)
+        recordableViewModel.snackBarObservable.subscribe {
+            snackBar.enqueue(
+                JFXSnackbar.SnackbarEvent(
+                    messages["noRecorder"],
+                    messages["addPlugin"].toUpperCase(),
+                    5000,
+                    false,
+                    EventHandler {
+                        audioPluginViewModel.addPlugin(true, false)
+                    })
+            )
+        }
+    }
+
+    private fun createAudioPluginProgressDialog() {
+        // Plugin active cover
+        val dialog = progressdialog {
+            root.addClass(AppStyles.progressDialog)
+            recordableViewModel.contextProperty.toObservable().subscribe { newContext ->
+                when (newContext) {
+                    TakeContext.RECORD -> graphic = AppStyles.recordIcon("60px")
+                    TakeContext.EDIT_TAKES -> graphic = AppStyles.editIcon("60px")
+                    else -> {}
+                }
+            }
+        }
+        recordableViewModel.showPluginActiveProperty.onChange {
+            Platform.runLater {
+                if (it) dialog.open() else dialog.close()
+            }
+        }
+    }
+
+    private fun Node.bindVisibleToDraggingNodeProperty() {
+        visibleProperty().bind(draggingNodeProperty.booleanBinding { it != null })
+    }
+
+    private fun startDrag(evt: StartDragEvent) {
+        if (evt.take != recordableViewModel.selectedTakeProperty.value) {
+            draggingNodeProperty.set(evt.draggingNode)
+            animateDrag(evt.mouseEvent)
+        }
+    }
+
+    private fun animateDrag(evt: AnimateDragEvent) {
+        animateDrag(evt.mouseEvent)
+    }
+
+    private fun animateDrag(evt: MouseEvent) {
+        if (draggingNodeProperty.value != null) {
+            val widthOffset = 348
+            val heightOffset = 200
+            dragContainer.toFront()
+            dragContainer.relocate(evt.sceneX - widthOffset, evt.sceneY - heightOffset)
```

Comment:
The magic numbers should probably be provided by the implementing subclasses.  Maybe introduce `abstract protected val dragOffsetX` (and Y).

Or, consider calculating these offsets here from dragContainer's current dimensions.

Reply:
I reworked how we calculate the position for the dragging node, so this should be unnecessary now.